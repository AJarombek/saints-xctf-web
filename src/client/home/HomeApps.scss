/*
 * Sass styling for the HomeApps component
 * @author Andrew Jarombek
 * @since 1/22/2020
 */

@import "../fonts";

.sxctf-home-apps {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #fdfdfd;
  padding: 75px 0;

  p {
    @include roboto-slab;
    font-size: 28px;

    @media screen and (max-width: 800px) {
      text-align: center;
      font-size: 24px;
    }
  }

  .app-stores {
    display: flex;
    flex-direction: row;

    figure {
      width: 75px;
      height: 75px;
      cursor: pointer;

      @media screen and (max-width: 420px) {
        width: 50px;
        height: 50px;
      }

      img {
        width: 100%;
        height: 100%;
        filter: grayscale(1) brightness(1);
        transition: filter 0.5s ease;

        &:hover {
          filter: none;
        }
      }
    }
  }
}
