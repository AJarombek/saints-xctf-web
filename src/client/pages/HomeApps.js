/**
 * Section of the home page listing the different apps that SaintsXCTF is available on.
 * @author Andrew Jarombek
 * @since 1/21/2020
 */

import React from 'react';

const HomeApps = () => {
    return (
        <div className="sxctf-home-apps">

        </div>
    );
};

export default HomeApps;
