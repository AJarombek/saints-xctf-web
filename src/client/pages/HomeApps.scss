/*
 * Sass styling for the HomeApps component
 * @author Andrew Jarombek
 * @since 1/22/2020
 */

.sxctf-home-apps {
  p {

  }
  .app-stores {

  }
}
