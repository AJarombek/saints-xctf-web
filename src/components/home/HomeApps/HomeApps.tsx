/**
 * Section of the home page listing the different apps that SaintsXCTF is available on.
 * @author Andrew Jarombek
 * @since 1/21/2020
 */

import React from 'react';

const HomeApps: React.FunctionComponent = () => {
  return (
    <div className="sxctf-home-apps">
      <p>SaintsXCTF has iOS and Android apps in these app stores</p>
      <div className="app-stores">
        <a href="#">
          <figure>
            <img src="/asset/app-store.png" alt="" />
          </figure>
        </a>
        <a href="#">
          <figure>
            <img src="/asset/google-play-store.svg" alt="" />
          </figure>
        </a>
        <a href="#">
          <figure>
            <img src="/asset/amazon-app-store.png" alt="" />
          </figure>
        </a>
      </div>
    </div>
  );
};

export default HomeApps;
