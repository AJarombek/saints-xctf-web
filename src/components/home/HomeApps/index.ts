import HomeApps from "./HomeApps";

export default HomeApps;
