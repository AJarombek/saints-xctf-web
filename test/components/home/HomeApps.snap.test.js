/**
 * Snapshot test for the HomeApps component.
 * @author Andrew Jarombek
 * @since 2/18/2020
 */

import React from 'react';
import HomeApps from '../../../src/components/home/HomeApps';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<HomeApps />).toJSON();
  expect(tree).toMatchSnapshot();
});
