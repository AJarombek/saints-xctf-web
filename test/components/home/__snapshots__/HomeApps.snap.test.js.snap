// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sxctf-home-apps"
>
  <p>
    SaintsXCTF has iOS and Android apps in these app stores
  </p>
  <div
    className="app-stores"
  >
    <a
      href="#"
    >
      <figure>
        <img
          alt=""
          src="/asset/app-store.png"
        />
      </figure>
    </a>
  </div>
</div>
`;
